ci-setups/python-gitlab.md

```diff
@@ -0,0 +1,78 @@
+# Python CI Setup Guide for GitLab
+
+## Table of Contents
+1. **Introduction**
+2. **Prerequisites**
+3. **CI Configuration Overview**
+4. **Defining the Pipeline**
+5. **Configuring Variables**
+6. **Pipeline Stages**
+    - Lint
+    - Test
+7. **Workflow Rules**
+8. **Updated `.gitlab-ci.yml` Example**
+9. **Conclusion**
+
+## Introduction
+This guide provides a detailed walkthrough for setting up a Continuous Integration (CI) pipeline for Python projects at Osmosys using GitLab. It emphasizes automating testing and linting to maintain high-quality code standards.
+
+## Prerequisites
+- A Python project hosted on GitLab.
+- Basic knowledge of Python development and GitLab CI/CD principles.
+
+## CI Configuration Overview
+The foundation of the CI pipeline is the `.gitlab-ci.yml` file placed at the project's root. This configuration file outlines the environment settings, stages, and jobs that constitute your CI pipeline.
+
+## Defining the Pipeline
+The pipeline comprises stages dedicated to linting code and running automated tests, essential practices for upholding code quality and verifying functionality.
+
+## Configuring Variables
+Variables offer a way to dynamically adjust the pipeline's settings. Key variables include:
+- `DEFAULT_IMAGE`: Specifies the Docker image for the pipeline.
+- `merge_request_branches`: Determines which branches trigger the pipeline upon receiving merge requests.
+
+## Pipeline Stages
+### Lint
+This stage employs tools like `flake8` to evaluate code quality.
+### Test
+During this stage, the pipeline runs automated tests using `pytest`, setting the `PYTHONPATH` to ensure modules are correctly found.
+
+## Workflow Rules
+The pipeline's execution is fine-tuned to activate only for merge requests that target certain branches, streamlining the CI process to focus on critical updates.
+
+## Updated `.gitlab-ci.yml` Example
+```yaml
+variables:
+  DEFAULT_IMAGE: "python:3.12"
+
+image: $DEFAULT_IMAGE
+
+stages:
+  - lint
+  - test
+
+before_script:
+  - pip install -r requirements.txt
+
+lint:
+  stage: lint
+  script:
+    - flake8 .
+  tags:
+    - python-3.12 # Ensure you are using correct runner
+
+test:
+  stage: test
+  script:
+    - export PYTHONPATH=$PYTHONPATH:$(pwd) # Ensure Python can locate the project modules
+    - pytest tests/
+  tags:
+    - python-3.12 # Ensure you are using correct runner
+
+workflow:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event" && $CI_MERGE_REQUEST_TARGET_BRANCH_NAME =~ /^main|development$/'
+```
```

Comment:
Kindly take inspiration from here 
https://github.com/OsmosysSoftware/dev-standards/blob/docs/python-gitlab-ci/ci-setups/angular-gitlab.md

We want to keep in same format.

This document lack the screenshot and detailed steps.

We also want to keep all documents to have same heading. So take all related heading from angular gitlab document.
Also the bonus section is not required.

Reply:
Partially done in https://github.com/OsmosysSoftware/dev-standards/pull/60/commits/993fbaad7e0ae662ee3709418d6a64e8b2824f43

Document, including the headings and content at some places has been re-formatted and written to maintain consistency.

`TODO`: Need to add screenshots and detailed steps for the usage of CI.

Reply:
Can we not do it as part of this PR?

Reply:
I right now dont feel the need to do that.